# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/IdentityCredentialStorageService.cpp
# Commit: e13a0519fa01
# Full Hash: e13a0519fa01a003cb6a67937fd5bb9b703b77e6
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2023-03-31 15:30:41
# Description:
#   Bug 1823655: Set IdentityCredentialStorageService.mShuttingDown if the init() is called during shutdown. r=pbz
#   
#   The mShuttingDown wasn't set in the init() function during shutdown. In
#   this case, we won't register ShutdownBlocker, so we won't flip
#   mShuttingDown. This could case shutdown hanging becasue
# ==============================================================================

diff -r ef7d2e82eb5a -r e13a0519fa01 toolkit/components/credentialmanagement/IdentityCredentialStorageService.cpp
--- a/toolkit/components/credentialmanagement/IdentityCredentialStorageService.cpp	Fri Mar 31 10:53:02 2023 +0000
+++ b/toolkit/components/credentialmanagement/IdentityCredentialStorageService.cpp	Fri Mar 31 11:05:15 2023 +0000
@@ -126,6 +126,8 @@
   AssertIsOnMainThread();
 
   if (AppShutdown::IsInOrBeyond(ShutdownPhase::AppShutdown)) {
+    MonitorAutoLock lock(mMonitor);
+    mShuttingDown.Flip();
     return NS_ERROR_ILLEGAL_DURING_SHUTDOWN;
   }
 
